chirp/drivers/baofeng_uv17Pro.py

```diff
@@ -0,0 +1,1427 @@
+            _upload(self)
+        except errors.RadioError:
+            raise
+        except Exception:
+            # If anything unexpected happens, make sure we raise
+            # a RadioError and log the problem
+            LOG.exception('Unexpected error during upload')
+            raise errors.RadioError('Unexpected error communicating '
+                                    'with the radio')
+
+    def get_features(self):
+        """Get the radio's features"""
+
+        rf = chirp_common.RadioFeatures()
+        rf.has_settings = True
+        rf.has_bank = False
+        rf.has_tuning_step = False
+        rf.can_odd_split = True
+        rf.has_name = True
+        rf.has_offset = True
+        rf.has_mode = True
+        rf.has_dtcs = True
+        rf.has_rx_dtcs = True
+        rf.has_dtcs_polarity = True
+        rf.has_ctone = True
+        rf.has_cross = True
+        rf.valid_modes = self.MODES
+        rf.valid_characters = self.VALID_CHARS
+        rf.valid_name_length = self.LENGTH_NAME
+        if self._gmrs:
+            rf.valid_duplexes = ["", "+", "off"]
+        else:
+            rf.valid_duplexes = ["", "-", "+", "split", "off"]
+        rf.valid_tmodes = ['', 'Tone', 'TSQL', 'DTCS', 'Cross']
+        rf.valid_cross_modes = [
+            "Tone->Tone",
+            "DTCS->",
+            "->DTCS",
+            "Tone->DTCS",
+            "DTCS->Tone",
+            "->Tone",
+            "DTCS->DTCS"]
+        rf.valid_skips = self.SKIP_VALUES
+        rf.valid_dtcs_codes = self.DTCS_CODES
+        rf.memory_bounds = (0, 999)
+        rf.valid_power_levels = self.POWER_LEVELS
+        rf.valid_bands = self.VALID_BANDS
+        rf.valid_tuning_steps = STEPS
+
+        return rf
+
+    def get_memory(self, number):
+        _mem = self._memobj.memory[number]
+
+        mem = chirp_common.Memory()
+        mem.number = number
+
+        if _mem.get_raw()[0] == 255:
+            mem.empty = True
+            return mem
+
+        mem.freq = int(_mem.rxfreq) * 10
+
+        if self._is_txinh(_mem):
+            # TX freq not set
+            mem.duplex = "off"
+            mem.offset = 0
+        else:
+            # TX freq set
+            offset = (int(_mem.txfreq) * 10) - mem.freq
+            if offset != 0:
+                if baofeng_common._split(self.get_features(), mem.freq, int(
+                          _mem.txfreq) * 10):
+                    mem.duplex = "split"
+                    mem.offset = int(_mem.txfreq) * 10
+                elif offset < 0:
+                    mem.offset = abs(offset)
+                    mem.duplex = "-"
+                elif offset > 0:
+                    mem.offset = offset
+                    mem.duplex = "+"
+            else:
+                mem.offset = 0
+
+        for char in _mem.name:
+            if str(char) == "\xFF":
+                char = " "  # The OEM software may have 0xFF mid-name
+            mem.name += str(char)
+        mem.name = mem.name.rstrip()
```

Comment:
You don't need to iterate for this. I think this whole loop is equivalent to:
```
mem.name = str(_mem.name).replace('\xFF', '').rstrip()
```
right?

Reply:
fixed. thnx